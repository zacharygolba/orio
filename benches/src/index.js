// @flow

// require('./line-counter')
// require('./pw-generator')
require('./simple-tasks')
