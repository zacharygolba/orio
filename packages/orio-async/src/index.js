// @flow

export * as sink from './sink'
export * as stream from './stream'
