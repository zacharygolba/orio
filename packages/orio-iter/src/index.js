// @flow

import type { Source } from 'orio-types'

import Iter from './iter'
import {
  createProducer,
  CharRange,
  Indexed,
  NumberRange,
  Repeat,
} from './producer'

export function chars(start?: string, end?: string): Iter<string> {
  const producer = new CharRange(start, end)
  return new Iter(producer)
}

export function from<T>(source?: Source<T>): Iter<T> {
  const producer = createProducer(source)
  return new Iter(producer)
}

export function of<T>(...items: $ReadOnlyArray<T>): Iter<T> {
  const producer = new Indexed(items)
  return new Iter(producer)
}

export function repeat<T>(value: T): Iter<T> {
  const producer = new Repeat(value)
  return new Iter(producer)
}

export function range(start?: number, end?: number): Iter<number> {
  const producer = new NumberRange(start, end)
  return new Iter(producer)
}
