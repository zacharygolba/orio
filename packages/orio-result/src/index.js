// @flow

export function done<T>(value?: T): IteratorResult<*, T> {
  return {
    done: true,
    value,
  }
}

export function next<T>(value: T): IteratorResult<T, *> {
  return {
    done: false,
    value,
  }
}
