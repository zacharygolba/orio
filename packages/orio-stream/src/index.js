// @flow

import * as sink from './sink'
import * as stream from './stream'

export { sink, stream }
