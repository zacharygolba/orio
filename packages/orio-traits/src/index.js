// @flow

export * from './as-iterator'
export { default as ToString } from './to-string'
