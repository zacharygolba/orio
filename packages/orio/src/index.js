// @flow

export * from 'orio-async'
export * as iter from 'orio-iter'
