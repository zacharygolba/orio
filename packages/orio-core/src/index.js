// @flow

import * as iter from 'orio-iter'

export * from 'orio-stream'
export { iter }
