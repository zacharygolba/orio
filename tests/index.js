/* @flow */

import * as iter from '../src'

test('#collect()', () => {
  const result = iter
    .range(1, 5)
    .collect()

  expect(result).toMatchSnapshot()
})

test('#enumerate()', () => {
  const result = iter
    .repeat('test')
    .take(5)
    .enumerate()
    .collect()

  expect(result).toMatchSnapshot()
})

test('#filter()', () => {
  const result = iter
    .range(1, 10)
    .filter(num => num % 2 === 0)
    .collect()

  expect(result).toMatchSnapshot()
})

test('#forEach', () => {
  const value = 'test'
  const result = iter
    .repeat(value)
    .take(5)
    .forEach(expect(value).toBe)

  expect(result).toBeUndefined()
})

test('#map()', () => {
  const result = iter
    .range(1, 10)
    .map(num => num ** 2)
    .collect()

  expect(result).toMatchSnapshot()
})

test('#next()', () => {
  const value = 'test'
  const subject = iter.repeat(value)

  expect(subject.next()).toMatchSnapshot()
  expect(subject.next()).toMatchSnapshot()
  expect(subject.next()).toMatchSnapshot()
})

test('#product()', () => {
  const a = iter
    .range(1, 5)
    .product()

  const b = iter
    .repeat('test')
    .take(5)
    .product()

  expect(a).toBe(2 * 3 * 4 * 5)
  expect(Number.isNaN(b)).toBe(true)
})

test('#sum()', () => {
  const a = iter
    .range(1, 5)
    .sum()

  const b = iter
    .repeat('test')
    .take(5)
    .sum()

  expect(a).toBe(1 + 2 + 3 + 4 + 5)
  expect(Number.isNaN(b)).toBe(true)
})


test('#join()', () => {
  const a = iter
    .repeat('test')
    .take(5)
    .join(' ')

  const b = iter
    .repeat('test')
    .take(5)
    .join()

  expect(a).toMatchSnapshot()
  expect(b).toMatchSnapshot()
})

test('#zip()', () => {
  const result = iter
    .range(0, 5)
    .zip(iter.range(5, 10))
    .collect()

  expect(result).toMatchSnapshot()
})

test('.from()', () => {
  const arr = [1, 2, 3]
  const str = 'test'
  const set = new Set(arr)
  const map = new Map(iter
    .repeat(str)
    .enumerate()
    .take(5)
    .collect())

  iter
    .from(arr)
    .zip(iter.range(0, 2).map(String))
    .forEach(([value, key]) => {
      expect(arr).toHaveProperty(key, value)
    })

  iter
    .from(str)
    .enumerate()
    .forEach(([idx, char]) => {
      expect(str.charAt(idx)).toBe(char)
    })

  iter
    .from(set)
    .map(set.has.bind(set))
    .forEach(expect(true).toBe)

  iter
    .from(map)
    .forEach(([key, value]) => {
      expect(map.get(key)).toBe(value)
    })
})

test('.range()', () => {
  expect(iter.range(1).collect()).toMatchSnapshot()
  expect(iter.range(1, 5).collect()).toMatchSnapshot()
  expect(iter.range(1, Infinity).take(10).collect()).toMatchSnapshot()
})

test('.repeat()', () => {
  expect(iter.repeat().take(5).collect()).toMatchSnapshot()
  expect(iter.repeat('test').take(5).collect()).toMatchSnapshot()
})
