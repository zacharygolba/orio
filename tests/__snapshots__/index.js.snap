// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#collect() 1`] = `
Array [
  1,
  2,
  3,
  4,
  5,
]
`;

exports[`#enumerate() 1`] = `
Array [
  Array [
    0,
    "test",
  ],
  Array [
    1,
    "test",
  ],
  Array [
    2,
    "test",
  ],
  Array [
    3,
    "test",
  ],
  Array [
    4,
    "test",
  ],
]
`;

exports[`#filter() 1`] = `
Array [
  2,
  4,
  6,
  8,
  10,
]
`;

exports[`#join() 1`] = `"test test test test test"`;

exports[`#join() 2`] = `"test,test,test,test,test"`;

exports[`#map() 1`] = `
Array [
  1,
  4,
  9,
  16,
  25,
  36,
  49,
  64,
  81,
  100,
]
`;

exports[`#next() 1`] = `
Object {
  "done": false,
  "value": "test",
}
`;

exports[`#next() 2`] = `
Object {
  "done": false,
  "value": "test",
}
`;

exports[`#next() 3`] = `
Object {
  "done": false,
  "value": "test",
}
`;

exports[`#zip() 1`] = `
Array [
  Array [
    0,
    5,
  ],
  Array [
    1,
    6,
  ],
  Array [
    2,
    7,
  ],
  Array [
    3,
    8,
  ],
  Array [
    4,
    9,
  ],
  Array [
    5,
    10,
  ],
]
`;

exports[`.range() 1`] = `
Array [
  1,
]
`;

exports[`.range() 2`] = `
Array [
  1,
  2,
  3,
  4,
  5,
]
`;

exports[`.range() 3`] = `
Array [
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
  10,
]
`;

exports[`.repeat() 1`] = `
Array [
  undefined,
  undefined,
  undefined,
  undefined,
  undefined,
]
`;

exports[`.repeat() 2`] = `
Array [
  "test",
  "test",
  "test",
  "test",
  "test",
]
`;
